File: flutter/test_app/lib/data/supabase/model/mapper/user_mapper.dart
import '../../../../domain/model/sportas_user.dart';
import '../DTOs/korisnik_dto.dart';
import '../DTOs/sportas_dto.dart';

SportasUser mapToDomain(KorisnikDto k, SportasDto s) {
  return SportasUser(
    id: k.idKorisnika,
    ime: k.imeKorisnika,
    prezime: k.prezimeKorisnika,
    datumRodenja: DateTime.parse(s.datumRodenja),
    tipClanarine: s.tipClanarine,
  );
}

File: kotlin_jetpack_compose/app/src/main/java/com/example/testapp/data/supabase/model/mappers/UserMapper.kt
package com.example.testapp.data.supabase.model.mappers

import com.example.testapp.data.supabase.model.DTOs.KorisnikDTO
import com.example.testapp.data.supabase.model.DTOs.SportasDTO
import com.example.testapp.domain.model.SportasUser
import java.time.LocalDate

fun korisnikISportasToDomain(
    korisnik: KorisnikDTO,
    sportas: SportasDTO
): SportasUser =
    SportasUser(
        id = korisnik.idKorisnika,
        ime = korisnik.imeKorisnika,
        prezime = korisnik.prezimeKorisnika,
        datumRodenja = LocalDate.parse(sportas.datumRodenja),
        tipClanarine = sportas.tipClanarine
    )